[VAULT-34485] UI: Internal namespace documentation updates (#31008)
hashicorp/vault: website/content/api-docs/system/internal-ui-namespaces.mdx

layout: api
page_title: /sys/internal/ui/namespaces - HTTP API
description: >-
  The `/sys/internal/ui/namespaces` endpoint exposes namespaces to the GUI.
---

# `/sys/internal/ui/namespaces`

The `/sys/internal/ui/namespaces` endpoint returns the namespaces that the current user has access to.

> This endpoint is used internally by the GUI and is an authenticated endpoint (requires a valid Vault token). You do not need to explicitly grant capabilities on the `/sys/internal/ui/namespaces` path in your policy. If your token has any access to a namespace, that namespace will be included in the response. For more information about namespace policies, see the [sys/namespaces API documentation](/vault/api-docs/system/namespaces).

Reviewer: ```suggestion
The namespaces endpoint is an authenticated endpoint used internally by the
Vault GUI to return all namespaces associated with the provided toekn. You do
not need to explicitly grant capabilities on the `/sys/internal/ui/namespaces`
path in your policy. For more information about namespace policies, see the
[sys/namespaces API documentation](/vault/api-docs/system/namespaces).
```
Style correction: write in active voice, avoid "this" as a pronoun

> Due to the nature of its intended usage, there is no guarantee on backwards compatibility for this endpoint.

Reviewer: ```suggestion
Internal endpoints are not intended for general use and make no guarantees on
backwards compatibility over time.
```
Style correction: write in active voice, avoid "this" as a pronoun

## Get namespaces

> This endpoint lists the namespaces relevant to the GUI.

Reviewer: ```suggestion
List namespaces associated with the provided Vault token for use in the GUI.
```

```shell-session
$ curl \
    --header "X-Vault-Token: ..." \
    --request GET \
    http://127.0.0.1:8200/v1/sys/internal/ui/namespaces
```

### Sample response

```json
{
    "request_id": "28a1ca80-cc97-913d-8ed2-844ea501ae0d",
    "lease_id": "",
    "renewable": false,
    "lease_duration": 0,
    "data": {
        "keys": [
            "parent/",
            "parent/child/",
            "parent/child/grandchild/",

>             "parent2/"

Reviewer 2: Docs folks may have an opinion here but I think we try to have "real world" names for sample code

Reviewer: Yeah, we generally try to have something meaningful for example paths. For secret paths, we've used things like `dev/sandbox/` and `dev/production`. For namespaces I don't think we've settled on an explicit pattern yet, but something that reflects common usage (like team hierarchies) would be appropriate. For example `software/` --> `software/eng` --> `software/eng/dev`, `software/eng/qc`
